components/brave_wayback_machine/brave_wayback_machine_tab_helper.h

```diff
@@ -6,49 +6,72 @@
 #ifndef BRAVE_COMPONENTS_BRAVE_WAYBACK_MACHINE_BRAVE_WAYBACK_MACHINE_TAB_HELPER_H_
 #define BRAVE_COMPONENTS_BRAVE_WAYBACK_MACHINE_BRAVE_WAYBACK_MACHINE_TAB_HELPER_H_
 
-#include <memory>
-
+#include "base/functional/callback_forward.h"
 #include "base/gtest_prod_util.h"
 #include "base/memory/raw_ref.h"
 #include "base/memory/weak_ptr.h"
+#include "brave/components/brave_wayback_machine/wayback_machine_url_fetcher.h"
+#include "brave/components/brave_wayback_machine/wayback_state.h"
 #include "content/public/browser/web_contents_observer.h"
 #include "content/public/browser/web_contents_user_data.h"
+#include "ui/gfx/native_widget_types.h"
 
-class BraveWaybackMachineDelegate;
 class PrefService;
 
 class BraveWaybackMachineTabHelper
     : public content::WebContentsObserver,
+      public WaybackMachineURLFetcher::Client,
       public content::WebContentsUserData<BraveWaybackMachineTabHelper> {
  public:
-  explicit BraveWaybackMachineTabHelper(content::WebContents* contents);
+  static void CreateIfNeeded(content::WebContents* web_contents);
+
+  using WaybackStateChangedCallback =
+      base::RepeatingCallback<void(WaybackState state)>;
+
   ~BraveWaybackMachineTabHelper() override;
 
   BraveWaybackMachineTabHelper(const BraveWaybackMachineTabHelper&) = delete;
   BraveWaybackMachineTabHelper& operator=(
       const BraveWaybackMachineTabHelper&) = delete;
 
-  void set_delegate(std::unique_ptr<BraveWaybackMachineDelegate> delegate);
+  void SetWaybackStateChangedCallback(WaybackStateChangedCallback callback);
+  void set_active_window(gfx::NativeWindow window) { active_window_ = window; }
+  gfx::NativeWindow active_window() const { return active_window_; }
```

Comment:
This also would be better if the bubble closes itself when tab closes or state changes.

Reply:
Done. will try to close whenever active tab changes.

Reply:
I think we can still move active_window() stuff from here to wayback_machine_state_manager() or bubble's static memory. I didn't find any usage of active_window in this component. And if we move these, we can keep this component independent from the ui stuff.

Reply:
Well... I think tab helper is good place for caching per-tab data as bubble is launched for each tab.
Instead of having them in one global var, I wanted to control it with more clear ownership.
So, can check whether this tab has launched bubble or not easily.

Reply:
Many `PageActionIconView::GetBubble()` is implemented in similar way.
Its bubble pointer is cached in its controller(`WebContentsUserData`) and retrieve it via its controller.

Reply:
Fair enough!